segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -1602,6 +1641,76 @@ private void read00(RevisionDataInput input, TruncationRecord.TruncationRecordBu
         }
     }
 
+    /**
+     * Journal record for segment append.
+     */
+    @Builder(toBuilder = true)
+    @Data
+    @EqualsAndHashCode(callSuper = true)
+    static class AppendRecord extends SystemJournalRecord {
```

Comment:
If there is a rollback when upgrading to this PR and some old Segment Store finds a system journal with an `AppendRecord`, would it be skipped or will make the initialization process to fail? can we add a test for that?

Reply:
good point. 
let me know how best to handle the roll back.

Reply:
https://github.com/pravega/pravega/pull/6909